language/move-lang/src/parser/ast.rs

```diff
@@ -211,6 +212,8 @@ pub type SpecBlockMember = Spanned<SpecBlockMember_>;
 // Specification condition kind.
 #[derive(PartialEq, Debug)]
 pub enum SpecConditionKind {
+    Assert,
```

Comment:
The SpecCondition was intended for pre/post conditions (that's why it is called "Condition", and invariants are a different kind of item). Should we pull this into a different item then?

Reply:
I interpreted invariants as something that are resource invariants or module invariants.  I found it cleaner to put together assert/assume together with other function-level specifications such as preconditions and postconditions, the unifying feature being that all of these are conditions that hold at specific syntactic positions in the code.  

Also note that I don't have a separate category called loop invariant.  I believe that we will be able to interpret assert/assume as asserted/assumed loop invariants just by virtue of the position where they occur.

Reply:
I'm fine you keep it like this, however, I think invariants, pre/post conditions, and assume/assert/loop invariants... are all different kinds of things. For instance, conditions can use `old`. Loop invariants may need other info than just the expression, we don't know yet.